tools/clang/lib/Headers/hlsl/dx/linalg.h

```diff
@@ -43,14 +43,24 @@ enum MatrixLayout {
 // Helper for signedness
 //
 namespace details {
-template <typename T> bool IsUnsigned() { return false; }
 
-#ifdef __HLSL_ENABLE_16_BIT
-template <> bool IsUnsigned<uint16_t>() { return true; }
-#endif
+template <typename T> struct IsUnsigned {
+  static const bool Value = false;
+};
+
+template <> struct IsUnsigned<uint32_t> {
+  static const bool Value = true;
+};
 
-template <> bool IsUnsigned<uint32_t>() { return true; }
-template <> bool IsUnsigned<uint64_t>() { return true; }
+template <> struct IsUnsigned<uint64_t> {
+  static const bool value = true;
+};
+
+#ifdef __HLSL_ENABLE_16_BIT
+template <> struct IsUnsigned<uint16_t> {
+  static const bool value = true;
+};
```

Comment:
These specializations use a different casing for `Value`.
```suggestion
template <> struct IsUnsigned<uint64_t> {
  static const bool Value = true;
};

#ifdef __HLSL_ENABLE_16_BIT
template <> struct IsUnsigned<uint16_t> {
  static const bool Value = true;
};
```

Reply:
Yeah, this is the issue Joshua mentioned in his review that I mentioned at today's morning meeting was a bug. I am adding high-level API tests that would have caught this.

Reply:
I added tests for uint and uint16_t, removed the uint64_t case since that will fail the builtin overload check.